pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -380,13 +384,21 @@ func (npc *NetworkPolicyController) ensureTopLevelChains() {
 		ensureRuleAtPosition(builtinChain, args, uuid, 1)
 	}
 
-	whitelistServiceVips := []string{"-m", "comment", "--comment", "allow traffic to cluster IP", "-d",
-		npc.serviceClusterIPRange.String(), "-j", "RETURN"}
-	uuid, err := addUUIDForRuleSpec(kubeInputChainName, &whitelistServiceVips)
+	whitelistPrimaryServiceVips := []string{"-m", "comment", "--comment", "allow traffic to primary cluster IP range",
+		"-d", npc.primaryServiceClusterIPRange.String(), "-j", "RETURN"}
+	uuid, err := addUUIDForRuleSpec(kubeInputChainName, &whitelistPrimaryServiceVips)
 	if err != nil {
 		klog.Fatalf("Failed to get uuid for rule: %s", err.Error())
 	}
-	ensureRuleAtPosition(kubeInputChainName, whitelistServiceVips, uuid, serviceVIPPosition)
+	ensureRuleAtPosition(kubeInputChainName, whitelistPrimaryServiceVips, uuid, serviceVIPPosition)
+
+	whitelistSecondaryServiceVips := []string{"-m", "comment", "--comment", "allow traffic to primary cluster IP range",
+		"-d", npc.secondaryServiceClusterIPRange.String(), "-j", "RETURN"}
+	uuid, err = addUUIDForRuleSpec(kubeInputChainName, &whitelistSecondaryServiceVips)
+	if err != nil {
+		klog.Fatalf("Failed to get uuid for rule: %s", err.Error())
+	}
+	ensureRuleAtPosition(kubeInputChainName, whitelistSecondaryServiceVips, uuid, serviceVIPPosition)
```

Comment:
This code should only execute if there is dual-stack. Otherwise `npc.secondaryServiceClusterIPRange.String()` is nil